cmake/modules/BuildBoost.cmake

```diff
@@ -167,6 +167,21 @@ function(do_build_boost version)
       URL_HASH SHA256=${boost_sha256}
       DOWNLOAD_NO_PROGRESS 1)
   endif()
+
+  foreach(c ${Boost_BUILD_COMPONENTS})
+    string(TOUPPER ${c} upper_c)
+    if(c MATCHES "^python")
+      set(c "python${Python3_VERSION_MAJOR}${Python3_VERSION_MINOR}")
+    endif()
+    if(Boost_USE_STATIC_LIBS)
+      set(Boost_${upper_c}_LIBRARY_OUT
+        ${boost_root_dir}/lib/${CMAKE_STATIC_LIBRARY_PREFIX}boost_${c}${CMAKE_STATIC_LIBRARY_SUFFIX})
+    else()
+      set(Boost_${upper_c}_LIBRARY_OUT
+        ${boost_root_dir}/lib/${CMAKE_SHARED_LIBRARY_PREFIX}boost_${c}${CMAKE_SHARED_LIBRARY_SUFFIX})
+    endif()
+    list(APPEND Boost_LIBRARIES_OUT ${Boost_${upper_c}_LIBRARY_OUT})
+  endforeach()
```

Comment:
could you restructure this part so we don't replicate it in the `build_boost` macro? if it's impossible / difficult to
```cmake
add_dependencies(Boost::${c} Boost)
```
before the `Boost` target is defined. probably we can collect `Boost::*` libraries in a list and claim the dependencies in another loop.

what do you think?

Reply:
The `build_boost` macro can depend directly on the `Boost_${upper_c}_LIBRARY_OUT` variables defined here instead of recomputing them, so we can just delete #235-241 removing the duplication.

Im not quite sure what you are suggesting, we can move all of the component target definitions from `build_boost` into this macro and `add_dependencies(Boost::${c} Boost)` will late bind to the ExternalProject_Add (at least on cmake 3.11.4). This looks like this patch https://gist.github.com/kkartaltepe/e57f55c6d67a96933ef487e89430651f

Reply:
@kkartaltepe sorry for the latency. i came up with #39826, as some of us mentioned that it would be better to move to ninja for faster compilation. could you help review it?

Reply:
Sure thanks ill close this PR out then

Reply:
thanks for your contribution!